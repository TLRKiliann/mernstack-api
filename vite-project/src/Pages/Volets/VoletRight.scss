/*VOLETS*/
/*Right volet*/

.volet--R {
  position: absolute;
  top: 0px;
  right: 0px;
  width: 400px;
  min-height: 100%;
  overflow: hidden;
}

@keyframes volet--rightdisplay {
  0% {
    transform: translateX(300px);
    overflow: hidden;
  }
  100% {
    transform: translateX(0px);
    overflow: hidden;
  }
}

.subvolet--R {
  position: relative;
  top: 0px;
  right: 0px;
  width: 400px;
  min-height: 120vh;
  background-color: royalblue;
  border-left: 1px solid blue;

  transform: translateX(0px);

  animation-name: volet--rightdisplay;
  animation-delay: 0s;
  animation-duration: 2s;
  animation-iteration-count: 1;
  animation-timing-function: ease;
  z-index: 5;
}

.subvolet--R button {
  margin: 10px 10px;
  width: 30px;
  height: 30px;

  border-radius: 50% 50%;
  background: dimgrey;
  color: lightgreen;
  &:hover {
    cursor: pointer;
    background: grey;
    color: whitesmoke;
  }
}

.subvolet--R p {
  margin: 20px 20px;
  padding: 5px 10px;
  text-align: center;
  background: lightblue;
  border: 2mm ridge royalblue;
  box-shadow: 0px 0px 4px whitesmoke;
  border-radius: 5px;
  color: royalblue;
}

.subvolet--R img {
  margin: -10px 0px;
  width: 60%;
  margin: auto;
  display: flex;
  background: royalblue;
  color: whitesmoke;
}

.section--voletr {
  margin: 20px 20px;
  padding: 20px 40px;
  background: lightblue;
  border: 2mm ridge royalblue;
  box-shadow: 0px 0px 4px whitesmoke;
  border-radius: 10px;
}

.li--right {
  margin-bottom: 10px;
  padding-bottom: 10px;
  border-bottom: 1px dashed blue;
  color: royalblue;
  &:hover {
    color: orange;
  }
}

.link--tagvoletr {
  text-decoration: none;
  font-weight: bold;
  color: royalblue;
}

.link--tagvoletr:hover {
  color: navy;
}


.volet--R2 {
  position: absolute;
  top: 0px;
  right: 0px;
  width: 400px;
  min-height: 100%;
  overflow: hidden;
}

.subvolet--R2 {
  position: relative;
  width: 400px;
  top: 0px;
  right: -400px;
  width: 400px;
  min-height: 120vh;
  background-color: royalblue;
  border-left: 1px solid blue;

  transform: translateX(0px);

  animation-name: volet--righthidden;
  animation-delay: 0s;
  animation-duration: 2s;
  animation-iteration-count: 1;
  animation-timing-function: ease;
  z-index: 5;
}

@keyframes volet--righthidden {
  0% {
    transform: translateX(-400px);
    overflow: hidden;
  }
  100% {
    transform: translateX(0px);
    overflow: hidden;
  }
}

.subvolet--R2 button {
  margin: 10px 10px;
  width: 30px;
  height: 30px;

  border-radius: 50% 50%;
  background: dimgrey;
  color: lightgreen;
  &:hover {
    cursor: pointer;
    background: grey;
    color: whitesmoke;
  }
}

.subvolet--R2 p {
  margin: 20px 20px;
  padding: 5px 10px;
  text-align: center;
  background: lightblue;
  border: 2mm ridge royalblue;
  box-shadow: 0px 0px 4px whitesmoke;
  border-radius: 5px;
  color: royalblue;
}

.subvolet--R2 img {
  margin: -10px 0px;
  width: 60%;
  margin: auto;
  display: flex;
  background: royalblue;
  color: whitesmoke;
}