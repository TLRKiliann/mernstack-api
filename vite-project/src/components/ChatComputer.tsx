import React, { useState, useEffect } from 'react'
import { useParams, useNavigate } from 'react-router-dom'
import { useAuthLogin } from '../context/AuthProvider'
import usePersonnalHook from '../hook/personnal.hook'
import serviceRouting from '../services/serviceRouting'
import { db_computers } from '../models/db_computers'
import { db_computeOne } from '../models/db_computeList'
import { db_computeTwo } from '../models/db_computeList'
import { db_computeThree } from '../models/db_computeList'
import { db_computeFour } from '../models/db_computeList'
import { db_computeFive } from '../models/db_computeList'
import { db_computeSix } from '../models/db_computeList'
import { db_computeSeven } from '../models/db_computeList'
import { db_computeEight } from '../models/db_computeList'
import { db_computeNine } from '../models/db_computeList'
import { db_computeTen } from '../models/db_computeList'
import { db_computeEleven } from '../models/db_computeList'
import { db_computeTwelve } from '../models/db_computeList'
import { computerType } from '../models/computerType'
import img_1 from '../assets/hdd.jpg'
import img_2 from '../assets/cpu.jpg'
import img_3 from '../assets/ram.jpg'
import img_4 from '../assets/motherboard.png'
import img_5 from '../assets/systemd.jpg'
import img_6 from '../assets/process.jpg'
import img_7 from '../assets/world_bin.png'
import img_8 from '../assets/reqres.jpg'
import img_9 from '../assets/firewall.jpg'
import img_12 from '../assets/Cybersecurity.jpg'
import './styleComponents/ChatComputer.scss'


const ChatComputer: React.FC = () => {

  const {id} = useParams<{id?: string}>();
  const { username } = useAuthLogin()
  const Navigate = useNavigate()
  
  const users = usePersonnalHook()
  
  //console.log(users, "- object users")

  const [userRoom, setUserRoom] = useState<Array<UserType>>([])
  //console.log(userRoom, 'userRoom state')
  const [computerDb, setComputerDb] = useState<string>("")
  const [imgBg, setImgBg] = useState<string>("")
  const [links, setLinks] = useState<Array<computerType>>([])

  const handleNavigation = (link: string) => {
    const myTimeOut = setTimeout(() => {
      Navigate(`/computerroom/${link}`)
    }, 1000)
  }

  const handleSetUserRoom = (link: string) => {
    const user = users?.find(user => user.firstName === username)
    const id: number = user.id;
    const changeConfRoom = {...user, isConnected: true}
    const addRoomUser = {
      id: user.id,
      img: user.img,
      firstName: username,
      lastName: user.lastName,
      age: user.age,
      email: user.email,
      location: user.location,
      gender: user.gender,
      mainroom: computerDb,
      room: link,
      isConnected: true,
      signalRecieve: user.signalRecieve,
      messagebox: user.messagebox,
      returnConfirm: user.returnConfirm
    }
    //console.log(addRoomUser.id, 'by id')
    if (user) {
      serviceRouting
        .updateRoomName(id, changeConfRoom)
        .then(initialUserRoom => {
          setUserRoom(users?.map(user => user.id === id ? {
            id: user.id,
            img: user.img,
            firstName: username,
            lastName: user.lastName,
            age: user.age,
            email: user.email,
            location: user.location,
            gender: user.gender,
            mainroom: computerDb,
            room: link,
            isConnected: true,
            signalRecieve: user.signalRecieve,
            messagebox: user.messagebox,
            returnConfirm: user.returnConfirm
            } : user
          ))
        })
        .catch((error) => {
          setUserRoom(users?.map(user => user.firstName !== username))
          alert(`Register name Room issue: ${user.firstName} not found !`)
        })
      handleNavigation(link)
    } else {
      alert("ERROR !!!")
    }
  }

  useEffect(() => {
    switch(id?.toString()) {
      case "1":
        setComputerDb("Hard Disk - SSD")
        setImgBg(img_1)
        setLinks(db_computeOne)
        break
      case "2":
        setComputerDb("CPU")
        setImgBg(img_2)
        setLinks(db_computeTwo)
        break
      case "3":
        setComputerDb("RAM")
        setImgBg(img_3)
        setLinks(db_computeThree)
        break
      case "4":
        setComputerDb("Drivers")
        setImgBg(img_4)
        setLinks(db_computeFour)
        break
      case "5":
        setComputerDb("Systemd")
        setImgBg(img_5)
        setLinks(db_computeFive)
        break
      case "6":
        setComputerDb("Process")
        setImgBg(img_6)
        setLinks(db_computeSix)
        break
      case "7":
        setComputerDb("Client-Server")
        setImgBg(img_7)
        setLinks(db_computeSeven)
        break
      case "8":
        setComputerDb("Protocols")
        setImgBg(img_7)
        setLinks(db_computeEight)
        break
      case "9":
        setComputerDb("Request - Response")
        setImgBg(img_8)
        setLinks(db_computeNine)
        break
      case "10":
        setComputerDb("Fire-Wall")
        setImgBg(img_9)
        setLinks(db_computeTen)
        break
      case "11":
        setComputerDb("Proxy - VPN")
        setImgBg(img_12)
        setLinks(db_computeEleven)
        break
      case "12":
        setComputerDb("Cyber-Security")
        setImgBg(img_12)
        setLinks(db_computeTwelve)
        break
      default:
        console.log("end of switch loop !")
        break
    }
  }, [])

  return(
    <div>
      
      <div data-testid="chatctestid" className="div--imgcompute">
        <img
          src={imgBg}
          width="100%"
          height="100%"
          className="img--chatcomputerroom"
          alt={imgBg}
        />
      </div>
      
      <div className="title--computerroom">
        <h1 className="compter--roomdb">{computerDb}</h1>
      </div>

      <div className='compter--room'>
        {Object.values(links).map((val) => (
          <h2 key={val.title} className="links--computer">

            <span
              onClick={() => handleSetUserRoom(val.link)} 
              className="span--linktoroom"
            >
              {val.link}
            </span>

          </h2>
          ))
        }
      </div>

    </div>
  )
}

export default ChatComputer;
