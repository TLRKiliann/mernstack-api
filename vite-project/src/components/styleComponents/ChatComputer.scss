/*
purple
orange
*/

.div--imgcompute {
  position: fixed;
  top: 0px;
  right: 0px;
  bottom: 0px;
  left: 0px;
  opacity: 0.7;
  background: linear-gradient(180deg, purple, orange);
  z-index: -1;

  .img--chatcomputerroom {
    opacity: 0.1;
  }
}

.title--computerroom {
  .compter--roomdb {
    margin-top: 120px;
    margin-left: 20px;
    font-family: "Acme", sans-serif;
    font-size: 2.4rem;
    color: whitesmoke;
    text-shadow: 2px 2px 1px purple,
    4px 3px 2px #242424;
  }
}

.computer--room {
  margin-top: 20px;
  
  .links--computer {
    font-family: "DejaVu Sans", sans-serif;
    cursor: pointer;

    .span--linktoroom {
      padding: 20px 0px;
      padding-left: 20px;
      margin: auto;
      margin-right: 20px;
      margin-left: 20px;
      display: flex;

      background-color: rgba(85, 45, 255, 0.6);
      border: 2mm outset rgba(85, 45, 255, 0.2);
      box-shadow: 0px 0px 4px slategrey;
      border-radius: 10px;

      text-decoration: none;
      color: whitesmoke;
      &:hover {
        background-color: rgba(85, 45, 255, 0.4);
        color: orange;
      }
      &:active {
        background-color: purple;
        color: lightgreen;
      }
    }
  }
}