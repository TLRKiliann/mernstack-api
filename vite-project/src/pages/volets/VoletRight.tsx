import React from 'react'
import { Link } from 'react-router-dom'
import mother from '../../assets/background/motherboard.png'
import './VoletRight.scss'

interface VoletRightProps {
  isOpenRight: boolean
  handleVoletsRight: () => void
}

const VoletRight: React.FC = (props: VoletRightProps) => {

  return(
    <div>
      {props.isOpenRight ? (
        <div className="volet--R">
          <div className="subvolet--R">
            <button 
              onClick={props.handleVoletsRight}
              className="btn--voletr"
            >
              X
            </button>
            <p>Chat Computing</p>

            <img
              src={mother}
              width='240px'
              height='140px'
              alt='no img volet-R 1'
            />

            <section className="section--voletr">
              {props.computers?.map((computer) => (
                <li key={computer.id} className="li--right">
                  <Link className="link--tagvoletr"
                    to={`/chatcomputer/${computer.id}`}
                  >
                    {computer.title}
                  </Link>
                </li>
                ))
              }
            </section>

          </div>
        </div>
        ) : (
        <div className="volet--R2">
          <div className="subvolet--R2">

            <button 
              onClick={props.handleVoletsRight}
              className="btn--voletr"
            >
              X
            </button>
            <p>Chat Rooms</p>

            <img
              src={mother}
              width='240px'
              height='140px'
              alt='no img volet-R 2'
            />

            <section className="section--voletr">
              {props.computers?.map((computer) => (
                <li key={computer.id} className="li--right">
                  <Link className="link--tagvoletr"
                    to={`/chatcomputer/${computer.id}`}
                  >
                    {computer.title}
                  </Link>
                </li>
                ))
              }
            </section>

          </div>
        </div>
        )
      }
    </div>
  )
}

export default VoletRight