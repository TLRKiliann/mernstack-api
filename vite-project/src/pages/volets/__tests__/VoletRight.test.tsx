import React from "react"
import { screen, render } from '@testing-library/react'
import { test, expect, vi } from 'vitest'
import { create } from 'react-test-renderer'
import '@testing-library/jest-dom'
import VoletRight from "../VoletRight"
import ChatComputer from "../VoletRight"

test('VoletRight MatchSnapShot test VoletRight', () => {
  const treeVoletR = create(<VoletRight />)
  expect(treeVoletR.toJSON()).toMatchSnapshot()
})

test('renders chat rooms link', () => {
  render(<VoletRight />)
  const linkElement = screen.getByText(/chat computing/i)
  expect(linkElement).toBeInTheDocument()
})

test("VoletR test ChatComputer", () =>{
  const treeChat = create(
    <ChatComputer />
  )
  expect(treeChat).toMatchSnapshot()
})

test('VoletR callMyLogo to be defined', () => {
  const callMyLogo = vi.fn()
  const funcLogo = callMyLogo
  expect(funcLogo).toBeDefined()
})

test('VoletR callMyLogo return id', () => {
  const callMyLogo = vi.fn()
  callMyLogo()
  expect(callMyLogo).toHaveBeenCalledTimes(1)
})
