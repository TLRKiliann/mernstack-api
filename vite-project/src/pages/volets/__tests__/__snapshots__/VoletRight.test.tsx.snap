// Vitest Snapshot v1

exports[`VoletR test ChatComputer 1`] = `
<div>
  <div
    className="volet--R2"
  >
    <div
      className="subvolet--R2"
    >
      <button
        className="btn--voletr"
      >
        X
      </button>
      <p>
        Chat Computing
      </p>
      <img
        alt="no img volet-R 2"
        height="140px"
        src="/src/assets/background/motherboard.png"
        width="240px"
      />
      <section
        className="section--voletr"
      />
    </div>
  </div>
</div>
`;

exports[`VoletRight MatchSnapShot test VoletRight 1`] = `
<div>
  <div
    className="volet--R2"
  >
    <div
      className="subvolet--R2"
    >
      <button
        className="btn--voletr"
      >
        X
      </button>
      <p>
        Chat Computing
      </p>
      <img
        alt="no img volet-R 2"
        height="140px"
        src="/src/assets/background/motherboard.png"
        width="240px"
      />
      <section
        className="section--voletr"
      />
    </div>
  </div>
</div>
`;
